cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -0,0 +1,405 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/array.hpp"
+#include "oneapi/dal/backend/memory.hpp"
+#include "oneapi/dal/backend/primitives/ndshape.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+enum class ndorder {
+    c, /* C-style ordering, row-major in 2D case */
+    f /* Fortran-style ordering, column-major in 2D case */
+};
+
+template <ndorder order>
+struct transposed_ndorder;
+
+template <>
+struct transposed_ndorder<ndorder::c> {
```

Comment:
It is not clear how to define 'transpose' operation for multi-dimensional arrays w/o permutation strategy. What means `c` and `f` orders for multi-dimensional arrays?

Reply:
Ok, I see how you define `c` and `f` in the `ndarray_base` ctor. It would be nice to add a definition of each order in a place where `ndorder` is defined.

Reply:
Hmm, there is description for c and f.